custom-ci-cd-environment-with-docker_5caf3f432c7d3a392f9ceb50.md

```diff
@@ -113,14 +129,47 @@ To enable caching support, the following requirements need to be met:
 
 To enable running Docker-in-Docker the `docker` executable needs to be installed.
 
-An example `Dockerfile` that meets the above criteria would be:
+### Building a minimal Docker image for Semaphore
+
+An example `Dockerfile` that meets the minimal requirements can be constructed
+with the following Dockerfile:
 
 ``` Dockerfile
 FROM ubuntu:18.04
 
 RUN apt-get -y update && apt-get install -y git lftp docker
 ```
 
+### Extending Semaphore's pre-build convenience Docker images
+
+An alternative to building a fully custom Docker image from scratch is to extend
+one of the pre-built images from [Semaphore's DockerHub repository][dockerhub-semaphore].
+
+For example, to extend one of Semaphore's Ruby based images and install MySQL
+libraries use the following Dockerfile:
+
+``` Dockerfile
+FROM semaphoreci/ruby:2.6.2
+
+RUN apt-get -y install -y mysql-client libmysqlclient-dev
+```
+
+### Optimizing Docker images for fast CI/CD
+
+To achieve the best CI/CD experience and fastest results make sure to follow
+these guidelines:
+
+- Pre-install all the tools and languages in your Docker image. Avoid
+  downloading and installing tools every time you run a job as it increases
+  build time.
+
+- Keep the size of the Docker images small to avoid unnecessary boot time of
```

Comment:
maybe link to https://semaphoreci.com/blog/2016/12/13/lightweight-docker-images-in-5-steps.html ?